sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -214,6 +214,8 @@ class CodeGenContext {
     case dt: DataType if isPrimitiveType(dt) => s"($c1 > $c2 ? 1 : $c1 < $c2 ? -1 : 0)"
     case BinaryType => s"org.apache.spark.sql.catalyst.util.TypeUtils.compareBinary($c1, $c2)"
     case NullType => "0"
+    case s: StructType if s.fields.forall(_.dataType.isInstanceOf[AtomicType]) =>
```

Comment:
@rxin after checking `CreateStructUnsafe`, I still feel confused.

Basically, `genComp` will generate a one line java code that will be evaluated in `if()`. Even in `CreateStructUnsafe`, its usage of `GenerateUnsafeProjection.createCode` generates a block of java code to run to set up the value of `GeneratedExpressionCode.primitive`.

So in `genComp`, for `StructType`, we still need to generate one line java code. I don't see how we can use the approach of `GenerateUnsafeProjection.createCode` to do this for `genComp`.